README.md

```diff
@@ -179,3 +179,20 @@ sbt publishLocal
 cd python
 pip install -e . # pip install -e .[all] to install all optional extras (see "Install from pypi")
 ```
+
+
+### Utilities
+
+[pre-commit](https://pre-commit.com/) can be helpful in keep consistent code format with the repository. 
```

Comment:
Lets rephrase this sentence to

```
[pre-commit](https://pre-commit.com/) is an optional tool for developers to keep the code style consistent.  After setup, it will automatically run various formatters and lints before a commit. 
```